state/azure/tablestorage/tablestorage.go

```diff
@@ -53,44 +55,56 @@ import (
 const (
 	keyDelimiter        = "||"
 	valueEntityProperty = "Value"
-	operationTimeout    = 1000
 
-	accountNameKey = "accountName"
-	accountKeyKey  = "accountKey"
-	tableNameKey   = "tableName"
+	accountNameKey  = "accountName"
+	accountKeyKey   = "accountKey"
+	tableNameKey    = "tableName"
+	cosmosDBmodeKey = "cosmosDBmode"
 )
 
 type StateStore struct {
 	state.DefaultBulkStore
-	table *storage.Table
-	json  jsoniter.API
+	client       *aztables.Client
+	json         jsoniter.API
+	cosmosDBmode bool
 
 	features []state.Feature
 	logger   logger.Logger
 }
 
 type tablesMetadata struct {
-	accountName string
-	accountKey  string
-	tableName   string
+	accountName  string
+	accountKey   string
+	tableName    string
+	cosmosDBmode bool
 }
 
-// Initialises connection to table storage, optionally creates a table if it doesn't exist.
+// Init Initialises connection to table storage, optionally creates a table if it doesn't exist.
 func (r *StateStore) Init(metadata state.Metadata) error {
 	meta, err := getTablesMetadata(metadata.Properties)
 	if err != nil {
 		return err
 	}
 
-	client, _ := storage.NewBasicClient(meta.accountName, meta.accountKey)
-	tables := client.GetTableService()
-	userAgent := "dapr-" + logger.DaprVersion
-	client.AddToUserAgent(userAgent)
-	r.table = tables.GetTableReference(meta.tableName)
+	cred, err := aztables.NewSharedKeyCredential(meta.accountName, meta.accountKey)
+	if err != nil {
+		return err
+	}
+	r.cosmosDBmode = meta.cosmosDBmode
+	var serviceURL string
+	if r.cosmosDBmode {
+		serviceURL = fmt.Sprintf("https://%s.table.cosmos.azure.com", meta.accountName)
```

Comment:
Wha if instead of using a boolean `cosmosDBmode` we allowed setting the endpoint directly? Right now we support only Cosmos DB and Azure Storage Tables running Azure public cloud. But there are other clouds (with different endpoints) and then there's Azure Stack etc.

Reply:
I'll separately add a custom endpoint option which will take precedence if set. I think that is a better approach.

Reply:
Sounds good to me!